browser/src/libs/code_intelligence/code_intelligence.tsx

```diff
@@ -477,213 +472,224 @@ export function handleCodeHost({
         observeOn(animationFrameScheduler)
     )
 
-    interface CodeViewState {
-        subscriptions: Subscription
-        roots: WorkspaceRootWithMetadata[]
+    const rootRefCounts = new Map<string, number>()
```

Comment:
Always document the semantics of primitive type arguments to collections

```suggestion
    /** Map from workspace URI to number of editors referencing it */
    const rootRefCounts = new Map<string, number>()
```